packages/showcases/css/stories/components/indicators/progressbar/examples/linear.html

```diff
@@ -207,13 +213,19 @@
     function addProgress(container, progress, label, value) {
       label.textContent = `${value}%`;
       container.setAttribute('aria-valuenow', `${value}`);
-      progress.style.transform = `translateX(calc(-100% + ${value}%))`;
+      progress.style.setProperty(
+        '--vtmn-progressbar_progress-transform',
+        `translateX(calc(${value}% - 100%))`,
+      );
     }
 
     function resetProgress(container, progress, label) {
       label.textContent = '0%';
       container.setAttribute('aria-valuenow', '0');
-      progress.style.transform = `translateX(-105%)`;
+      progress.style.setProperty(
+        '--vtmn-progressbar_progress-transform',
+        `translateX(-100%)`,
```

Comment:
```suggestion
        `translateX(-105%)`,
```

The 5% added make the linear progressbar really empty at 0%.

Reply:
But "5%" doesn't really make sense and is a [magic number](https://css-tricks.com/magic-numbers-in-css/), it will add more confusion I think. Why 105 ? Why not 103 or 110 ?

Various alternatives are possible here, but first: why is the progressbar not empty on 0%? Because the `stroke-with` property of our line makes it a little bigger on each side. My guess is that if the `stroke-with` is set to X px, then it adds x/2 on both sides. We could make it explicit:
```
`translateX(calc(-0.5rem / 2 - 100%))`

// or even
`translateX(calc(-var(--vtmn-progressbar_progress-stroke-width) / 2 - 100%))`
```

Another possibility could be to set `stroke-with` as a variable, for instance : 
```
progress.style.setProperty(
        '--vtmn-progressbar_progress-stroke-width',
        progress === 0 ? "0" : "0.5rem"
)
```

Another possibility could be to ignore this thin line at the beginning when the progress is 0. But then it should be validated by designers I guess.

What do you think @GaspardMathon ?

Reply:
I am more for the first option as it could be implemented in CSS easier.

Reply:
So at this time, we can either let the property set to 100% or we can reopen an issue to fix the design if it is a problem, what do you think @thibault-mahe ?

Reply:
Did you see my last message and commit ? I managed the issue with (what I think is) a more elegant fix. Thanks to the scale value of a transform, I can show / hide the visible progress. I managed to put the scale to 0 when progress is 0, or to 1 when progress is > 0.

You can see it here for instance: https://github.com/Decathlon/vitamin-web/blob/fix/linear-progressbar-animation/packages/showcases/css/stories/components/indicators/progressbar/examples/linear.html#L226

Reply:
Outdated